blas/tpls/KokkosBlas2_gemv_tpl_spec_decl.hpp

```diff
@@ -786,4 +786,111 @@ KOKKOSBLAS2_CGEMV_ROCBLAS(Kokkos::LayoutRight, Kokkos::Experimental::HIPSpace,
+      true, ETI_SPEC_AVAIL> {                                                  \
+    using execution_space = Kokkos::Experimental::SYCL;                        \
+    using device_type     = Kokkos::Device<execution_space, MEM_SPACE>;        \
+    using mem_traits      = Kokkos::MemoryTraits<Kokkos::Unmanaged>;           \
+    using AViewType =                                                          \
+        Kokkos::View<const SCALAR**, LAYOUT, device_type, mem_traits>;         \
+    using XViewType =                                                          \
+        Kokkos::View<const SCALAR*, LAYOUT, device_type, mem_traits>;          \
+    using YViewType = Kokkos::View<SCALAR*, LAYOUT, device_type, mem_traits>;  \
+                                                                               \
+    static void gemv(const execution_space& exec, const char kk_trans[],       \
+                     typename AViewType::const_value_type& alpha,              \
+                     const AViewType& A, const XViewType& X,                   \
+                     typename YViewType::const_value_type& beta,               \
+                     const YViewType& Y) {                                     \
+      bool row_major       = std::is_same<Kokkos::LayoutRight, LAYOUT>::value; \
+      const std::int64_t M = A.extent(0);                                      \
+      const std::int64_t N = A.extent(1);                                      \
+      oneapi::mkl::transpose trans = mode_kk_to_onemkl(kk_trans[0]);           \
+      const std::int64_t LDA       = row_major ? A.stride(0) : A.stride(1);    \
+      std::string label            = "KokkosBlas::gemv[TPL_ONEMKL," +          \
+                          Kokkos::ArithTraits<SCALAR>::name() + "]";           \
+      sycl::event event;                                                       \
+                                                                               \
+      Kokkos::Profiling::pushRegion(label);                                    \
+      using mag_type = kokkos_to_std_type_map<                                 \
+          SCALAR, Kokkos::ArithTraits<SCALAR>::is_complex>::type;              \
+      const mag_type* a = reinterpret_cast<const mag_type*>(A.data());         \
+      const mag_type* x = reinterpret_cast<const mag_type*>(X.data());         \
+      mag_type* y       = reinterpret_cast<mag_type*>(Y.data());               \
+      if (row_major) {                                                         \
+        event = oneapi::mkl::blas::row_major::gemv(                            \
+            exec.sycl_queue(), trans, M, N, alpha, a, LDA, x, 1, beta, y, 1);  \
+      } else {                                                                 \
+        event = oneapi::mkl::blas::column_major::gemv(                         \
+            exec.sycl_queue(), trans, M, N, alpha, a, LDA, x, 1, beta, y, 1);  \
```

Comment:
@lucbv I am unsure of this one.  The oneMKL gemv returned `event`, but I don't wait on that because I don't want to synchronize the GPU.  But I saw you did that in SpMV.  In other words,  the `event` variable is unused here.

Reply:
I believe in the case of SpMV I need to wait because the handle would otherwise be destroyed before the `release` function is called on it.
The SpMV TPL path will be optimized a bit for the handle destruction to be done in a non-blocking way but I have not worked on that part yet...

Reply:
Pipeline was clean, but anyway I removed the use of `event`.

Reply:
Thanks : )
Let us retest and this should be good